npeps/npep-133-fqdn-egress-selector.md

```diff
@@ -0,0 +1,428 @@
+1. Targeting in-cluster endpoints with FQDN selector is not recommended. There
+   are other selectors which can more precisely capture intent. However, if
+   in-cluster endpoints are selected:
+   *  ✅︎ Supported:
+      *  Selecting Pods using their [generated DNS
+         record](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pods)
+         (for example `pod-ip-address.my-namespace.pod.cluster.local`). This is
+         analogous to selecting the Pod by its IP address using the Network
+         selector.
+      * Headless Services can be selected using their [generated DNS
+        record](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services)
+        because the generated DNS records contain a list of all the Pod IPs that
+        back the service.
+   *  ❌ Not Supported:
+      * ClusterIP Services can not be selected using their [generated DNS
+        record](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services)
+        (for example `my-svc.my-namespace.svc.cluster.local`). This is
+        consistent with the behavior when selecting the Service VIP using the
+        Network selector.
+      * ExternalName Services return a `CNAME` record. See the entry below
+        about CNAME support.
+      * Any record which points to the IPs used for `LoadBalancer` type
+        services. This includes the `externalIPs` and the
+        `.status.loadBalancer.ingress` fields
+1. If the specified domain in a FQDN selector resolves to a [CNAME
+   record](#cname-records) the behavior of the implementor depends on the
+   returned response.
+
+   If the upstream resolver used [CNAME chasing](#cname-chasing) to fully
+   resolve the domain to a A/AAAA record and returns the resulting chain, the
+   implementor can use this information to allow traffic to the specified IPs.
+   However the implementor does not need to perform their own CNAME chasing or
+   to understand resolutions across multiple DNS requests.
+
+   For example, if the FQDN selector is allowing traffic to `www.kubernetes.io`:
+     *  If a DNS query to the upstream resolver returns *a single response* with
+        the following records:
+        ```
+        www.kubernetes.io  --  CNAME to kubernetes.io
+        kubernetes.io      --  A     to 1.2.3.4
+        ```
+        The implementor can use this response to allow traffic to `1.2.3.4`
+     *  If DNS query only responds with a CNAME record, the resolver is not
+        required to allow traffic even if subsequent requests resolve the full
+        chain:
+        ```
+        # REQUEST 1
+
+        www.kubernetes.io  --  CNAME to kubernetes.io
+        
+        # REQUEST 2
+
+        kubernetes.io      --  A     to 1.2.3.4
+        ```
+        The implementer can still deny traffic to `1.2.3.4` because no single
+        response contained the full chain required to resolve the domain.
+
+## Alternatives
```

Comment:
Other alternative is implement DNS filtering,  as cisco umbrella https://docs.umbrella.com/deployment-umbrella/docs/point-your-dns-to-cisco or  dnsfilter do ...